client/state/data-layer/wpcom/sites/comments/index.js

```diff
@@ -160,16 +174,19 @@ export const addComments = ( { dispatch }, { query: { siteId, status } }, { comm
 			status,
 			tree: [],
 		} );
+		dispatch( updateCommentsQuery( siteId, [], { offset, postId, status } ) );
 		return;
 	}
 
+	dispatch( updateCommentsQuery( siteId, comments, { offset, postId, status } ) );
+
 	const byPost = groupBy( comments, ( { post: { ID } } ) => ID );
 
-	forEach( byPost, ( postComments, postId ) =>
+	forEach( byPost, ( postComments, post ) =>
 		dispatch(
 			receiveComments( {
 				siteId,
-				postId: +postId, // keyBy => object property names are strings
+				postId: +post, // keyBy => object property names are strings
```

Comment:
+post? Can we have a more explicit type conversion, for example `parseInt( post, 10 )